### PR TITLE
UI: refactor transform

### PR DESCRIPTION
**Pull request by deanlee**
1. Only calculate transform once at initialization (`ui_nvg_init`)
2. Transform to screen coordinates in function car_space_to_full_frame.
3. Make sure the lead car indicator  is always displayed.

### CODE REVIEW

File: selfdrive/ui/paint.cc [hunk 1]
```diff
@@ -70,10 +71,12 @@ static void draw_chevron(UIState *s, float x_in, float y_in, float sz,
                           NVGcolor fillColor, NVGcolor glowColor) {
   float x, y;
   if (!car_space_to_full_frame(s, x_in, y_in, 0.0, &x, &y)) {
-    return;
+    if (x < 0 || y < 0) 
+      return;
   }
 
   sz = std::clamp((sz * 30) / (x_in / 3 + 30), 15.0f, 30.0f);
+  y = std::min(s->scene.viz_rect.bottom() - sz / 2,  y);
```
Comment (deanlee) [inline selfdrive/ui/paint.cc:79]: since y is already a screen coordinate, we can make sure it's always displayed here

**Comment by deanlee** [comment]
looks ok on my device, but not tested on the road,I'm not sure that the real environment will also work.

we already applied room by `nvgScale(s->vg, zoom, zoom);` can this solve the problem you said?

**Comment by pd0wm** [comment]
With the default zoom level you don't see the issue. Try setting the zoomlevel to something like 1 and make sure the lane lines run all the way to the edge of the screen.

**Comment by deanlee** [comment]
Yes, I've seen the problem.any further suggestions?

**Comment by deanlee** [comment]
It seems that we need the nearest point outside the screen

**Comment by pd0wm** [comment]
> It seems that we need the nearest point outside the screen

Yes. Just keeping one point outside the screen would also fix it.

**Comment by deanlee** [comment]
the problem fixed now

**Comment by deanlee** [comment]
Still use margins to simplify the problem

**Comment by pd0wm** [comment]
I understand the margin is not the cleanest solution, but keeping the first point outside the screen needed a lot of non trivial code. So agree with reverting back.

**Comment by deanlee** [comment]
I think the best part of this PR is that the indicator of the lead car can always be displayed.It will not disappear if the lead car is too close like the previous version.It’s clearer and safer for users

**Pull request merged**